### ISSUE TITLE
TranslationZed 0.9 bug

### ISSUE THREAD

**Issue by Mishasama** [TheIndieStone/ProjectZomboidTranslations#171]
Check this.
![020104181232](https://user-images.githubusercontent.com/54498172/71763297-dc9c2880-2f1d-11ea-9eb5-33d183b278b7.png)

**Comment by Mishasama** [comment]
Still can't delete the dropped string by this tool?

**Comment by Mishasama** [comment]
And it will delete all these comments and empty lines.
![020105092644](https://user-images.githubusercontent.com/54498172/71773260-8373d980-2f9d-11ea-9eb8-ed41bbc6aa39.png)

**Comment by Mishasama** [comment]
And the "TAB" before strings will be replaced with 4 spaces.

**Comment by Mishasama** [comment]
The SurvivalGuide_EN.txt text format is incompatible. It will be deleted all lines without a key.

**Comment by Fantazmagore** [comment]
The sorting function doesn't work after the searchbar is used (The empty selection is changed to "value" or "key")

**Comment by Fantazmagore** [comment]
I started to make a latvian translation. Created a folder "LV" where needed. TranslationZed created the language file named language.txtLV.txt

Inside the file I had to change the syntax manually. It was:

languageLV= {

key = "Value";
}

So the game recognised the file only after I named it correctly and got rid of the curly brackets and quotes.

**Comment by Fantazmagore** [comment]
And Translate Zed doesn't seem to understand that I have changed something. Everytime I open the program all files are shown green.

**Comment by Faalagorn** [comment]
Following my discussion with @Mishasama in #416 and @Fantazmagore at Discord, it turned out that TranslationZed actually *is* open sourced since May last year, with the code available at @timbaker's (EasyPickings) GitHub repo along with other tools at https://github.com/timbaker/translatoid, thanks!

It seems that it have been updated to 0.20 since then. I have figure out how to compile it yet, but precompiled binaries are available under https://github.com/timbaker/translatoid/releases.

Currently it seems TranslationZed uses the 4-space indents (and converts all whitespaces including tabs) and removes all comments (lines beginning with `--`) upon saving and it sorts files alphabetically like @Mishasama mentioned here and did in #416.

It also indeed seems not to understand and don't parse/ignore and remove everything after `..` for multi-line entries most notably used in SurvivalGuide as @Mishasama mentioned: https://github.com/TheIndieStone/ProjectZomboidTranslations/blob/2bed666afba833f907dc39263abc75cae122387d/EN/SurvivalGuide_EN.txt#L4

I'm not sure if you can remove strings and I'm not sure how the program handles removed strings, but it still feature a tool for old format of Items and Recipes (I personally didn't use them). 

It also seems that while EN UK appears there when loaded (regular EN does not), it doesn't save it properly, first similar issue @Fantazmagore experienced – language file is malformed, it also lacks support for all of the newly added .txt files as well (`credits.txt`, `description.txt` and `title.txt`). Due to odd format of the EN UK files it also produces different header (`_EN UK = {`), but I've yet have to see British English translation in-game and I already mentioned to @nsKpz in Discord that it probably should be `ENGB` similar to `PTBR` (no space) and to be closer to IANA/IEFT `en-GB`).

The tool is not ideal and in fact very crude, especially compared to online solution we have (Weblate, Crowdin, Transifex…), but it's what there currently and serves its purpose. Probably I won't have time to dig deeper into it today, but I will when I'll have some time to look into the issues and maybe fork it with appropriate PRs sent. :)

**Comment by Faalagorn** [comment]
> And Translate Zed doesn't seem to understand that I have changed something. Everytime I open the program all files are shown green.

I think you need to right click each string and choose "Mark As Resolved" option for it to save it when exiting.

**Comment by Fantazmagore** [comment]
> 

> 
> 
> > And Translate Zed doesn't seem to understand that I have changed something. Everytime I open the program all files are shown green.
> 
> I think you need to right click each string and choose "Mark As Resolved" option for it to save it when exiting.

Yup, I'm doing that. I change the line, click "Mark As Resolved", the entry becomes white. After I'm done for the day I click File->Save Translations and it creates the .txt file with the Resolved entries. And I close TranslationZED.

When I open TranslationZED and load the "Translate" folder next time, everything in LV (in my case), is green again. I figured that I could just continue translating from where I left off. Found the line alphabetically and continued, Marked the Resolved ones and saved afterwards. That was a big mistake. The program overwrote my previous progress and just created a new .txt from scratch. Ragequited because of this last weekend. :D

For now it looks like I'll better continue to do it manually, using Eclipse or Notepad.

Thanks, Faalagorn :)

**Comment by Faalagorn** [comment]
> When I open TranslationZED and load the "Translate" folder next time, everything in LV (in my case), is green again. I figured that I could just continue translating from where I left off. Found the line alphabetically and continued, Marked the Resolved ones and saved afterwards. That was a big mistake. The program overwrote my previous progress and just created a new .txt from scratch. Ragequited because of this last weekend. :D

That sounds aweful indeed… I spend too much time doing thing too and losing them can be heartbreaking, glad you're not giving up! I'll let you know if anything changes, but It's probably best to do it manually for you, as I think TranslatoidZed doesn't cope with new languages well at this point…

**Comment by Poltergeist-ix** [comment]
> Following my discussion with @Mishasama in #416 and @Fantazmagore at Discord, it turned out that TranslationZed actually _is_ open sourced since May last year, with the code available at @timbaker's (EasyPickings) GitHub repo along with other tools at https://github.com/timbaker/translatoid, thanks!

~~There is no license in the repository~~
It is updated with the MIT license.
